app/src/App.js

```diff
@@ -1,52 +1,42 @@
 import React from "react";
 
-import { useAuth0 } from "@auth0/auth0-react";
-
 import "./styles.css";
-// import reportWebVitals from "./reportWebVitals";
+import { withAuthenticationRequired, useAuth0 } from "@auth0/auth0-react";
+import { createBrowserHistory } from "history";
+import { Router, Route, Switch } from "react-router-dom";
+
+import Header from "./components/Header";
+import Home from "./components/Home";
+import PlantRecommendations from "./components/PlantRecommendations";
 import UserProfile from "./components/UserProfile";
+export const history = createBrowserHistory();
 
-const App = () => {
-  const {
-    loginWithRedirect,
-    logout,
-    user,
-    isAuthenticated,
-    isLoading,
-  } = useAuth0();
+const ProtectedRoute = ({ component, ...args }) => (
+  <Route component={withAuthenticationRequired(component)} {...args} />
+);
```

Comment:
Interesting is this your react router from react-router-dom? Does this make all the components sent into it a route ?

Reply:
this is an Auth0 protected route with react-router-dom, if you're using both i can send you the documentation. This was really difficult for me to get going. For Auth0 routes, you have to maintain authentication. We have to pass the args in an Auth0 method.